online-workflow/src/main/java/eu/itesla_project/online/StateAnalyzer.java

```diff
@@ -109,59 +91,59 @@ private void initStatus(){
+        OnlineTaskType currentStatus = OnlineTaskType.SAMPLING;
 
         try {
             // create new state
             logger.info("Analyzing state {}", stateId);
-            String stateIdStr=String.valueOf(stateId);
+            String stateIdStr = String.valueOf(stateId);
             context.getNetwork().getStateManager().cloneState(StateManager.INITIAL_STATE_ID, stateIdStr);
             context.getNetwork().getStateManager().setWorkingState(stateIdStr);
             // sample
             logger.info("{}: sampling started", stateId);
             status.put(currentStatus, OnlineTaskStatus.RUNNING);
-            stateListener.onUpdate(stateId, status,context.timeHorizon);
-            if ( !parameters.analyseBasecase() || stateId > 0 )
+            stateListener.onUpdate(stateId, status, context.timeHorizon);
+            if (!parameters.analyseBasecase() || stateId > 0)
                 sampler.sample();
             else
                 logger.info("{}: state = basecase", stateId);
             status.put(currentStatus, OnlineTaskStatus.SUCCESS);
-            stateListener.onUpdate(stateId, status,context.timeHorizon);
+            stateListener.onUpdate(stateId, status, context.timeHorizon);
             logger.info("{}: sampling terminated", stateId);
 
             // complete state with loadflow
-            currentStatus=OnlineTaskType.LOAD_FLOW;
+            currentStatus = OnlineTaskType.LOAD_FLOW;
             status.put(currentStatus, OnlineTaskStatus.RUNNING);
-            stateListener.onUpdate(stateId, status,context.timeHorizon);
+            stateListener.onUpdate(stateId, status, context.timeHorizon);
             logger.info("{}: loadflow started", stateId);
             LoadFlowResult result = loadFlow.run();
-            status.put(currentStatus, result.isOk()?OnlineTaskStatus.SUCCESS:OnlineTaskStatus.FAILED);
-            stateListener.onUpdate(stateId, status,context.timeHorizon);
+            status.put(currentStatus, result.isOk() ? OnlineTaskStatus.SUCCESS : OnlineTaskStatus.FAILED);
+            stateListener.onUpdate(stateId, status, context.timeHorizon);
             logger.info("{}: loadflow terminated", stateId);
-            if ( result.getMetrics() != null ) {
+            if (result.getMetrics() != null) {
                 logger.info("{}: loadflow metrics: {}", stateId, result.getMetrics());
-                if ( !result.getMetrics().isEmpty() )
+                if (!result.getMetrics().isEmpty())
                     onlineDb.storeMetrics(context.getWorkflowId(), stateId, OnlineStep.LOAD_FLOW, result.getMetrics());
             }
-            status.put(currentStatus, result.isOk()?OnlineTaskStatus.SUCCESS:OnlineTaskStatus.FAILED);
+            status.put(currentStatus, result.isOk() ? OnlineTaskStatus.SUCCESS : OnlineTaskStatus.FAILED);
 
-            if ( parameters.storeStates() ) {
+            if (parameters.storeStates() || ((parameters.analyseBasecase()) && (stateId == 0))) {
```

Comment:
Do we really want to store the basecase state if the storeStates parameters is false? If the answer is yes, we should rename this parameter to something like "storePostContingencyStates" to be more explicit.